src/main/drivers/pwm_output.c

```diff
@@ -328,12 +315,21 @@ void motorDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8
         IOConfigGPIO(motors[motorIndex].io, IOCFG_AF_PP);
 #endif
 
+        /* standard PWM outputs */
         if (useUnsyncedPwm) {
-            const uint32_t hz = timerMhzCounter * 1000000;
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, hz / motorConfig->motorPwmRate, idlePulse, motorConfig->motorPwmInversion);
-        } else {
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, 0xFFFF, 0, motorConfig->motorPwmInversion);
+            pwmRate = motorConfig->motorPwmRate;
         }
+        const uint32_t clock = timerClock(timerHardware->tim);
+        const uint16_t prescaler = ceilf((clock / pwmRate) / 65535.0f); /* used to find the desired timer frequency for max resolution */
```

Comment:
Just add (divisor - 1) before division, truncation in integer divide will do the rest (works only for positive numbers) 

Btw:is 65535 correct? It is a bit unusual, I would expect 2^16..